### ISSUE TITLE
Angular 2 release plan

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#4549]
This ticket is to plan and manage Angular 2 release as per original thread here https://github.com/jhipster/generator-jhipster/issues/3664 (which is locked for core team only)

discussion started https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/jhipster-dev/nyworLcz6MQ/4ghzZklkCAAJ

Any further discussion on this matter should happen here

The steps are in order

**Step 1:**

- [x] Migrate everything from the app generator to NG2 (Means we dont have any ng1 controller, directive and service) and this is 98% done we only have few config files and interceptors to finish
- [x] Merge webpack (I'm 90% ok with the PR #4422 needs some polish regarding my comments and also need to open a specific ticket with things to do)
- [x] Merge to master (I would like to do this as there is gonna be quite a lot of conflicts and stuff, I already tried locally once so I know what are the things to expect)
- [x] Refactor from `/client`  and `/client-2` to `/client/templates/angularjs` and `/client/templates/angular` (I will do this, and its very important that we do this architecturally now rather than later)
- [x] Add a error log for entity generator to let people know that it doesn't yet work with ng2 (If you guys are patient to wait for say one more week then I can migrate this as well, but this can be done only after merge with master)
- [x] Fix atleast Karma tests (Personally I wouldn't like to ship a software with broken tests) protractor can wait as its an option
- [x] Freeze all development on 3.x and move it to maintenance mode (It should only get very important update or fix if required)
- [x] Release "JHipster 4.0.0" notice that angular2 support will be marked **BETA**

**Step 2:** We could go with multiple 4.0.x patch releases for this or can do a single minor with all below together 

- [x] Migrate entity sub generator if its not already done for above release
- [x] Remove ng1 dependencies from the generated angular2 app and make it work as a pure ng2 app (This should improve performance etc)
- [x] Migrate to BS4 and improve UI (for alpha the current UI with BS3 would do eventhough it has some glitches)
- [ ] Improve webpack config to add in missing steps from the gulp workflow
- [x] Fix protractor tests if not already done
- [x] Fix tslint warnings
- [ ] Improve performance, fix known/reported bugs and polish
- [ ] finish items from #4794 

**Step 3:** 

- [ ] Remove beta label from angular2 option (when all critical bugs known/reported are fixed, all dependencies updated to atleast RCs)
- [ ] Make angular 2 the default option in the question
- [x] Test all scenarios and make sure there is no bugs https://github.com/jhipster/generator-jhipster/issues/4301

**Comment by fmatuszewski** [comment]
I would like to help with weboack.

**Comment by deepu105** [comment]
Yes you can take a look at  #4422

**Comment by deepu105** [comment]
This is the progress checklist from issue #3664 so that we can continue marking things as done here
- [x] Create a client-2 sub gen, which is a copy of the client sub-gen, and do all the dev there
- [x] Introduce a new question to use `client-2` instead of `client` so that users can choose what they want -> with "client" they will stay on AngularJS 1 (and have less risk), with "client-2" they follow our migration path, and will end up with AngularJS 2 app in the end (but more work/risk for them). All work is now done on "client-2"
- [x] Integrate Typescript
- [x] Integrate SystemJS
- [x] Integrate angular2
- [x] setup gulp to process and lint typescript
- [x] modularize the client code to common, admin, account and entity
- [x] serve generated js and html files from target folder
- [x] Move to [ng-bootstrap](https://ng-bootstrap.github.io)
- [ ] Move to Bootstrap 4
- [x] export all the ts files and import them in appropriate modules
- [ ] add all other deps via npm (and probably drop Bower)
- [x] use the upgrader component to upgrade ng1 directives / controllers
- [x] upgrade the services, controllers and directives to be fully angular2 for the main generator
- [x] Migrate to component router or ui-router-ng2 instead of ui-router
- [ ] Set html5 mode as default url strategy 
- [ ] create `entity-2` sub generator to use when project is using `client-2`
- [ ] upgrade the services, controllers and directives to be fully angular2 for the `entity-2` sub-generator
- [ ] polish templates
- [ ] check [styleguide](https://angular.io/docs/ts/latest/guide/style-guide.html) compliance
- [ ] optimize the way how files are served from target|build folder
- [ ] add ts compile task to pom/gradle build
- [ ] optimize prod gulp build
- [ ] once stable enough deprecate `client` and make `client-2` default (major version release)
#### progress of individual module migrations
- [x] Account: activate (@flaviencathala/@wmarques)
- [x] Account: password (@flaviencathala)
- [x] Account: register (@flaviencathala)
- [x] Account: reset (@flaviencathala)
- [x] Account: sessions (@ruddell)
- [x] Account: settings (@ruddell)
- [x] Account: social
- [x] Admin: audits 
- [x] Admin:configuration
- [x] Admin:docs
- [x] Admin:health
- [x] Admin:logs
- [x] Admin:metrics (@deepu105)
- [x] Admin:tracker (@deepu105)
- [x] Admin:gateway (@ruddell)
- [x] Admin:user-management (@ruddell)
- [ ] Blocks:config
- [x] Blocks:handlers
- [ ] Blocks:interceptors
- [x] Components:alert (@wmarques)
- [x] Components:auth (@deepu105)
- [x] Components:form (@deepu105) all moved to shared
- [x] Components:language
- [x] Components:login (@flaviencathala)
- [x] Components:profiles
- [x] Components:util  (@deepu105) all moved to shared
- [x] Home
- [x] Layouts:error (@deepu105)
- [x] Layouts:navbar (@deepu105)
- [x] shared:all  (@deepu105)

The known issues checklist has been moved to #4407 to make this thread lighter

If anyone is looking for a guide to use the angular-2 branch here it is https://github.com/jhipster/generator-jhipster/blob/angular-2/NG2-GUIDE.md

**Comment by deepu105** [comment]
@pascalgrimaud would need your help to add some travis builds for angular2 once we merge the branch to master (which should be soon) we atleast need

- angular 2 default config
- angular 2 with jwt + gradle
- angular 2 with sass and no translation

what do you think?

**Comment by pascalgrimaud** [comment]
@deepu107 : of course I will work on Travis build for Angular2. Im ok with your options, so it will be 4 new builds

**Comment by deepu105** [comment]
isnt it 3? how many slots do we have left?

**Comment by pascalgrimaud** [comment]
4 left. I thought sass and no translation separately

**Comment by deepu105** [comment]
I think they can be combined. we can save the one slot for something in
future


Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
@jhipster/developers @flaviencathala @VictorADS I have removed all ng1 dependencies and angular1 from the app. now we have a pure ng2 app :smile_cat: 
We still have few more config files which either needs to be removed or we need to find something equivalent in ng2 but those are minor.
we still need to do some polish

**Comment by deepu105** [comment]
@sendilkumarn @wmarques can you guys take a look and see if anything can be improved.
@christopherthielen would you be able to help to do a high level review and suggest if something can be improved? I think we are in a good shape now

**Comment by wmarques** [comment]
@deepu105 I check this tonight and let you know my feedback

**Comment by mraible** [comment]
@deepu105 I tried creating an application with the latest branch and running it and it has come webpack compilation errors. 

```
[INFO] ERROR in ./src/main/webapp/app/shared/auth/base64.service.ts
[INFO] Module parse failed: /Users/mraible/dev/jhipster-ng2-latest/node_modules/angular2-template-loader/index.js!/Users/mraible/dev/jhipster-ng2-latest/node_modules/awesome-typescript-loader/dist/entry.js!/Users/mraible/dev/jhipster-ng2-latest/node_modules/tslint-loader/index.js!/Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts 'return' outside of function (61:4)
[INFO] You may need an appropriate loader to handle this file type.
[INFO] |         enc1 = enc2 = enc3 = enc4 = '';
[INFO] |     }
[INFO] |     return output;
[INFO] | }
[INFO] | decode(input);
[INFO]  @ ./src/main/webapp/app/shared/index.ts 13:9-41
[INFO]  @ ./src/main/webapp/app/app.module.ts
[INFO]  @ ./src/main/webapp/app/app.main.ts
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/blocks/interceptor/http.provider.ts:16:23
[INFO] Cannot find name 'LocalStorageService'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/blocks/interceptor/http.provider.ts:17:25
[INFO] Cannot find name 'SessionStorageService'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/blocks/interceptor/http.provider.ts:36:8
[INFO] Cannot find name 'LocalStorageService'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/blocks/interceptor/http.provider.ts:37:8
[INFO] Cannot find name 'SessionStorageService'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/auth-oauth2.service.ts:7:34
[INFO] Cannot find module '../tracker/tracker.service'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/auth-oauth2.service.ts:29:40
[INFO] Cannot find name 'base64'. Did you mean the instance member 'this.base64'?
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/auth-oauth2.service.ts:45:15
[INFO] A function whose declared type is neither 'void' nor 'any' must return a value.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:6:4
[INFO] Unexpected token. A constructor, method, accessor, or property was expected.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:12:4
[INFO] Cannot find name 'encode'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:12:12
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:12:19
[INFO] ';' expected.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:23:19
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:24:19
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:25:19
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:26:19
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:51:4
[INFO] Cannot find name 'decode'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:51:4
[INFO] Unreachable code detected.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:51:12
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:51:19
[INFO] ';' expected.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:63:8
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:63:16
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:65:19
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:66:34
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:67:34
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:68:34
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:69:34
[INFO] Cannot find name 'input'.
[INFO]
[INFO] ERROR in [default] /Users/mraible/dev/jhipster-ng2-latest/src/main/webapp/app/shared/auth/base64.service.ts:90:0
[INFO] Declaration or statement expected.
[INFO] Child html-webpack-plugin for "index.html":
[INFO]         + 21 hidden modules
[INFO] Child extract-text-webpack-plugin:
[INFO]         + 7 hidden modules
[INFO] Child extract-text-webpack-plugin:
[INFO]         + 5 hidden modules
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 31.535 s
[INFO] Finished at: 2016-12-05T06:29:52-07:00
[INFO] Final Memory: 49M/350M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.0:webpack (webpack build) on project jhipster: Failed to run task: 'webpack.js --config webpack/webpack.dev.js' failed. (error code 2) -> [Help 1]
```

Here's my `.yo-rc.json`:

```json
{
  "generator-jhipster": {
    "jhipsterVersion": "3.9.1",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "useSass": false,
    "applicationType": "monolith",
    "angularVersion": "angular2",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```

**Comment by cbornet** [comment]
@mraible I think oauth2 is not supported yet. Can you try with http session auth ?

**Comment by deepu105** [comment]
Yes to be frank I havent tested any other config except the default options. My priority is to get below working first

1. Default config
2. Default config with Gradle

**Comment by Bhushan001** [comment]
@deepu105  Hi Deepu, I tried running this project , all went well except there is no gulpfile & it tries to run gulpfile. So it throws error. previously I was able to run it using gulp but with webpack what  am I supposed to do?

**Comment by deepu105** [comment]
@Bhushan001 it shouldnt run gulp stuff we dont have that code anymore here. there might be something wrong in your setup, probably you didnt run `npm link` in the cloned project and within generated project. anyways you would have to figure it out as this is unreleased and we are really short handed to do support at the moment. hope you understand. If you find something wrong feel free to do PRs

**Comment by mraible** [comment]
@cbornet It works with http session auth. FWIW, I spent some more time trying to get Karma + webpack + typescript working, but failed.

**Comment by wmarques** [comment]
I tested it. Got an error first because I disabled the translation so he couldn't find the `jhi-translate` directive.
Once I enabled translation everything was fine, awesome work guys :+1:

**Comment by allahbaksh** [comment]
@deepu105, Are we on Yarn from angular2 branch. I think we have not merged that part. We are thinking to use jHipster for one of our new project with angular2. 
It would be helpful if it is merged.

**Comment by deepu105** [comment]
We will be merging it probably this week end to master

Thanks & regards,
Deepu

**Comment by Bhushan001** [comment]
@deepu105  Hi deepu , I am getting this error : `Can't bind to 'jhi-translate' since it isn't a known property of 'span'`

**Comment by wmarques** [comment]
@Bhushan001 this error occurs when translation is disabled, there should be a "jhi-translate" directive on a HTML template that shouldn't be there ;) First, we focus on the default options ( = translation enabled)

**Comment by pascalgrimaud** [comment]
@deepu105 : I begin to work on Travis build. As soon as the Angular-2 is merged into master, I will rebase my branch and make a PR for it

**Comment by deepu105** [comment]
Thanks man that would be great


Thanks & Regards,
Deepu

**Comment by Bhushan001** [comment]
@wmarques  I tried enabling translation in my yo.rc.json .still getting the same error.

**Comment by deepu105** [comment]
@pascalgrimaud could you please help with the travis build please

**Comment by deepu105** [comment]
I have no idea why all protractor builds are failing in master

**Comment by pascalgrimaud** [comment]
@deepu105 : yes, already work in progress !

**Comment by jdubois** [comment]
@deepu105 is everything already merged?? I thought it would last all week-end!

**Comment by deepu105** [comment]
Nope it went better than I thought :smile:

**Comment by deepu105** [comment]
And I even refactored everything to the new `client/templates/${framework}` model and refactored existing angular1 client file writing to the much more efficient and friendly meta data structure :smiling_imp: 

Now if all of @jhipster/developers can play with it and give me feedback it would be nice :)
The protractor tests are failing but I didnt touch them so I have no idea what went wrong. Could be coz we upgraded protractor can someone help to check that please?

**Comment by pascalgrimaud** [comment]
@deepu105 : just added 3 new Travis builds. I already see some issues and try to fix them.

**Comment by pascalgrimaud** [comment]
After some tests:

Angular1:
- default app: when starting the application with `./mvnw`, there is a full build (including npm/yarn install, bower and gulp), so very slow start in dev profile
- when choosing noi18n, the user can't log into the application

Angular2:
- yarn is not well supported when choosing angular2
- noi18n not working: see https://travis-ci.org/jhipster/generator-jhipster/jobs/182863636

**Comment by deepu105** [comment]
I have fixed remaining Travis issues so now its green :)

**Comment by deepu105** [comment]
Did anybody get the karma tests working?

**Comment by Bhushan001** [comment]
@deepu105  is angular2 branch deleted? I am not able to take checkout of angular-2 branch.

**Comment by pascalgrimaud** [comment]
@Bhushan001 : the angular2 branch has been merged into master

**Comment by Bhushan001** [comment]
@pascalgrimaud are there any specific instructions to setup an angular2 app? I tried running yo jhipster , but its throwing 

> No Module named 'yeoman generator'

any ideas?

**Comment by pascalgrimaud** [comment]
Just follow https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md#-generator-development-setup

**Comment by deepu105** [comment]
guys please note that the webpack tasks in `package.json` have been renamed to `webpack:dev` and `webpack:prod`

**Comment by dimeros** [comment]
Suddenly, I am not able to login using the default settings ( http session authentication). When trying to login as admin:admin nothing happens.

**Comment by ramzimaalej** [comment]
@deepu105 does master has bs4 now?

**Comment by deepu105** [comment]
yes


Thanks & Regards,
Deepu

**Comment by erkieh** [comment]
Has someone already started on the entity-2 subgenerator?

**Comment by jdubois** [comment]
No, just mention on the ticket that you want to work on it, and you can do it. One of the main issues for me is that we haven't selected a "datatable" library at the moment, at my current client we have selected [swimlane/angular2-data-table](https://github.com/swimlane/angular2-data-table) but I don't know if it's going to work fine with Spring Data (for sorting/pagination)

**Comment by deepu105** [comment]
I would rather follow our current simple table implementation so that end
users can choose their own grid/table framework as there are quite a lot of
options


Thanks & Regards,
Deepu

**Comment by erkieh** [comment]
I'll start to work on the entity-2 sub-generator. I'm not sure if I'll get all the options (specifically pagination types) working since I am time constrained at the moment.

**Comment by deepu105** [comment]
@jhipster/developers Jhipster app has a new look take a look
![fireshot capture 1 - jhipster - http___localhost_9000_ _](https://cloud.githubusercontent.com/assets/1107223/21243820/e2884188-c31a-11e6-9bec-e9069a9c13be.png)

**Comment by wmarques** [comment]
Nice ! I preferred the white navbar but it's a detail ! Good job :+1:

**Comment by delverdev** [comment]
![e2884188-c31a-11e6-9bec-e9069a9c13be](https://cloud.githubusercontent.com/assets/7104443/21257906/24c044ca-c383-11e6-9995-e1a827e44d29.png)
I would:
- Move the logo to the left and the menu to the right.
- Move the content slightly to the left so that margin from the big logo's shoulder will be the same.
- Increase a margin between menu items.
- Change 'Jhipster' to 'JHipster' in the navbar logo.
- Make menu item's font weight bolder.
- Change triangles to chevrons (`^` upside down).
- Make languages shown like a flags (when a user that doesn't know app's preferred language will come to the site, I don't think he will 100% guess that flag icon is the language menu).

**Comment by deepu105** [comment]
@delverdev yes some of it can be done. 

1. We can move menu to right but moving logo left would make it hidden under the dev ribbon so something needs to be done. I guess we can make the dev ribbon 50% transparent
2. Yes can
3. Yes can
4. Its the app name so depends on what you choose during generation and hence cant be changed
5. yes can
6. Yes if it doesnt require tweaking bootstrap css too much
7. I quite dont understand what you mean here

Would you like to a PR with the suggestions?

**Comment by deepu105** [comment]
@pascalgrimaud I started with white navbar but then everything was too white and dull

**Comment by jdubois** [comment]
@delverdev @deepu105 don't use flags for languages, it's not a good way to display languages (some countries share the same languages, some countries have several languages...)

**Comment by pascalgrimaud** [comment]
@deepu105 : you mean @wmarques, not me
I'm happy with the colors :p

**Comment by delverdev** [comment]
@deepu105, I don't think I have much experience to PR, but  I could make a Gist.
> moving logo left would make it hidden under the dev ribbon

Yes, but client doesn't matter because there is no rev ribbon in production. Or indeed we could make it transparent.<br>
I read about flags for languages and now agree that it's not a best practice.

**Comment by wmarques** [comment]
@delverdev it's very easy to PR and make modification on generated files ;) See [here](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
Thanks for your inputs anyway :+1:

**Comment by delverdev** [comment]
Okay, I will try it tomorrow.

**Comment by deepu105** [comment]
@jhipster/developers @flaviencathala @erkieh I have started the entity generator migration to ng2. The below are the work done and TODO. I might not have time to finish it today and will be bit busy next week so if someone want to finish the TODO items please go ahead. Just make a comment here so that I know what you are working on

- [ ] Split entity generator client and server code
- [ ] Split client code to support angular and angularjs(1.x)
- [ ] Migrate the file writing to follow the new meta syntax
- [ ] Create typescript & html files for angular 2 in the` templates/client/angular` folder
- [ ] write basic tests
- [ ] write more tests for other options
- [ ] enable travis testing for entity
- [ ] migrate entity service
- [ ] migrate entity state
- [ ] migrate entity component and html
- [ ] migrate entity karma tests
- [ ] migrate entity protractor tests
- [ ] Fix the needle injections when ng2 is used

@jdubois once this is done I think we will have a release candidate

**Comment by deepu105** [comment]
the easiesnt way to migrate is to follow the user-management module as its similar

**Comment by erkieh** [comment]
@deepu105 
my WIP is in https://github.com/erkieh/generator-jhipster.  I managed to generate entities for my use case from the pre split/refactor state. Please feel free to discard all or reuse parts of my work done thus far because I probably don't have the time to get it to a state where I would do a PR.

**Comment by deepu105** [comment]
@erkieh its your work and hence I would like you to get the commit credit. Let me see if I can get your commits in

**Comment by pascalgrimaud** [comment]
@deepu105 :
- I will enable travis testing for entity as soon as it's ready
- I didnt follow closely: is protractor working for ng2? If yes, I can already enable it in Travis!

**Comment by deepu105** [comment]
@erkieh I have created a new branch from your branch so that your commit is retained and you get the deserved credit. Guys if anyone wants to work on top of this please use the entity-ng2 branch. here is a reference PR to visualize the changes https://github.com/jhipster/generator-jhipster/pull/4713

**Comment by deepu105** [comment]
@pascalgrimaud the last i checked it was working on master without entity. can you try and confirm locally as i'm little short on time

**Comment by deepu105** [comment]
and i really need some rest as well :sob:

**Comment by deepu105** [comment]
@pascalgrimaud the protractor tests for ng1 is failing in master. pretty sure i did some silly mistake but im exhausted would be grateful if you can take a look

**Comment by pascalgrimaud** [comment]
Same as you, I can work and have a look only tomorrow morning, before day work :-)
But no worry, we will repair it

**Comment by pascalgrimaud** [comment]
I commented the code here: https://github.com/jhipster/generator-jhipster/commit/b1dc23ca6ef6353139a74957a8a14db850f6d3e8#diff-fddb513f7a2937b33eb1c1f6a71e64a7

**Comment by anjia0532** [comment]
angular 4 beta is comming 
https://github.com/angular/angular/blob/master/CHANGELOG.md?utm_content=buffer0c5e3&utm_medium=social&utm_source=facebook.com&utm_campaign=buffer#400-beta0-2016-12-15

**Comment by deepu105** [comment]
What the hell happened 3.x ?

Thanks & regards,
Deepu

**Comment by wmarques** [comment]
@deepu105 Because the router is already at 3.x.x so to get every angular modules to the same versions, they skip 3.x and go straight to 4

**Comment by deepu105** [comment]
​really!!! ya thats not dumb at all :P​


Thanks & Regards,
Deepu

**Comment by m00g33k** [comment]
@deepu105 @wmarques Not exactly the reason. They're moving into semver which means you can also expect Angular 5 next year.

**Comment by cbornet** [comment]
@m00g33k I think everybody in JHipster is aware of semver. It's just that their versioning strategy is so confusing and messed up that it becomes funny !

**Comment by wmarques** [comment]
@m00g33k Agree with that, and it's a good thing that they want to be more clear about their versions :) But jumping from 2 to 4 is kinda funny as @cbornet said :D

**Comment by seawenzhu** [comment]
Hi, 
Is there a definite time about angular2 release plan?

**Comment by ramzimaalej** [comment]
what's left guys, and for which module do you need help ?

**Comment by deepu105** [comment]
@ramzimaalej would appreciate help to migrate and fix remaining app karma tests and the karma tests for entity on the ng2 branch take a look at #4302 or you could also help with #4407 or #4301

**Comment by ramzimaalej** [comment]
@deepu105 I don't see the branch ng2. Do you have the commit hash to restore it?

**Comment by pascalgrimaud** [comment]
@ramzimaalej : the old ng2 branch has been merged into master

**Comment by gmarziou** [comment]
Maybe an additional task unless you want it for after the release:

- [ ] add some needles for modules, for instance in `vendor.ts` and `vendor.css`

**Comment by jdubois** [comment]
@deepu105 I think we can close this now, do you agree?

**Comment by cloud4288** [comment]
@jdubois can I use this in production development now? I want to migrate my AngularJS project.

**Comment by jdubois** [comment]
@cloud4288 still in beta, but already a lot of people use it. Sorry but this ticket is not the right place for chatting.

**Comment by deepu105** [comment]
yes i'll move some todo from here to
https://github.com/jhipster/generator-jhipster/issues/4794 and will close it


Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
closing this as all TODO will be tracked in #4794
